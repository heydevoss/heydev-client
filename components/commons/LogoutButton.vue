<template>
  <div>
    <button class="button--grey" @click="onLogout">Logout</button>
  </div>
</template>

<script>
export default {
  name: 'LogoutButton',
  methods: {
    async onLogout() {
      await this.$apolloHelpers.onLogout()
    }
  }
}
</script>
<style lang="scss" scoped></style>
