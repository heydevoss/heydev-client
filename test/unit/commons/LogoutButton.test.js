import { shallowMount } from '@vue/test-utils'
import LogoutButton from '@/components/commons/LogoutButton.vue'

describe('<LogoutButton/>', () => {
  const onLogout = jest.fn()
  const wrapper = shallowMount(LogoutButton, {
    mocks: {
      $apolloHelpers: {
        onLogout
      }
    },
    stubs: {
      'el-button': '<button>Logout</button>'
    }
  })

  it('shoulbe be a Vue instance and render correctly', () => {
    expect(wrapper.isVueInstance()).toBeTruthy()
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should contain button', () => {
    expect(wrapper.html()).toContain('Logout')
    expect(wrapper.html().includes('<button>'))
  })

  it('should redirect when clicked', () => {
    wrapper.trigger('click')
    // expect(onLogout.mock.calls.length).toBe(1)
  })
})
