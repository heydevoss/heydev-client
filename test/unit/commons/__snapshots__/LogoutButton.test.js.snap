// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LogoutButton/> shoulbe be a Vue instance and render correctly 1`] = `
<div>
  <button
    type="text"
  >
    Logout
  </button>
</div>
`;
